lm_eval/api/task.py

```diff
@@ -316,17 +315,19 @@ def _process_doc(self, doc):
         """
         return doc
 
-    def create_choices(self, doc):
-        if self._config.create_choices is None:
+    def doc_to_choice(self, doc):
```

Comment:
Oh is this connected to the issue the other day. Yeah good catch. I'll adapt create_choices to process the `answer_choices`

Reply:
After re-checking promptsource, `answer_choices` doesn't need to be exposed because applying the prompt automatically creates both the input and target strings. Do you think it's important if we still expose `answer_choices`?

Reply:
Currently, if we use a Promptsource template with a multiple choice dataset, it apparently throws an error if you don't define `template_aliases: "{% set answer_choices = ... %}"` in the YAML (link: https://discord.com/channels/729741769192767510/755950983669874798/1128794360574656594 ) so we need to define behavior for the function `doc_to_choice()` if a promptsource template is being used.

Reply:
Huh, I see. 

Okay will make a fix for this.

Reply:
Okay added `qnli` to showcase how using `use_prompt` is enough to replace `doc_to_text`, `doc_to_target` and `doc_to_choice`.